test/autorest/complexgroup/zz_generated_polymorphic_helpers.go

```diff
@@ -103,7 +103,7 @@ func unmarshalMyBaseTypeClassification(rawMsg *json.RawMessage) (MyBaseTypeClass
 	}
 	var b MyBaseTypeClassification
 	switch m["kind"] {
-	case MyKindKind1:
+	case string(MyKindKind1):
```

Comment:
So it looks like we have tests in the test server that would have caught this.  This looks to be used in the `FlattenComplexClient`.  Can you please add those tests?

Reply:
The route is returning a 404 in the testserver, asking about it. Will see why that happened.

Reply:
I believe this operation was added to the swagger back in 2018 but never had a corresponding route added in the testserver. I think since 1. We arent really focusing on flattening here and 2. Our test is about enums, we should add another test in the testserver for this specific scenario. 

I've added an issue here: https://github.com/Azure/autorest.go/issues/606

Reply:
Sounds good.